### PR TITLE
Introduce responsive container block to render chart variants

### PR DESCRIPTION
**Pull request by kadamwhite**
Based in part on https://github.com/humanmade/datavis-block/pull/27 by @smccafferty, adapted to satisfy these design constraints

- Chart should be able to be removed from a responsive container and still render as expected
- Container-wrapped charts should respond to container dimensions, not viewport dimensions

chart resizing

### CODE REVIEW

**Review by goldenapples — APPROVED** [review]
Looks good, just left some notes from readthrough...

File: inc/blocks.php [hunk 1]
```diff
@@ -16,48 +16,109 @@ function bootstrap() : void {
...
+}
+
+/**
+ * Render a data-attribute with a specified value, IF that value is present.
+ *
+ * @param string $attribute_name Name of attribute, e.g. data-max-width.
+ * @param any    $value          Value of attribute.
+ */
+function maybe_render_attribute( string $attribute_name, $value ) : void {
+	if ( ! empty( $value ) ) {
```
[the remark this answers is not in the document]
Comment (kadamwhite) [reply inc/blocks.php:73]: I considered that, but given the limited set of uses of this helper, empty is equivalent to unneeded for all cases

File: src/blocks/responsive-container/index.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+		<InnerBlocks.Content />
+	);
+};
+
+export const settings = {
+	// Apply the block settings from the JSON configuration file.
+	...blockData,
+
+	transforms: {
+		from: [
```
[the remark this answers is not in the document]
Comment (kadamwhite) [reply src/blocks/responsive-container/index.js:248]: Block transforms do not take effect automatically—they must be invoked by a user by clicking a button
image

File: src/blocks/responsive-container/index.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+ * @returns {string} Rendered label for the specified breakpoint.
+ */
+const getBreakpointDescription = ( breakpoints, chartId ) => {
+	if ( breakpoints.length === 1 ) {
+		return __( 'Default visualization', 'vegalite-plugin' );
+	}
+
+	const lowerBound = breakpoints[ chartId ] || 0;
+	const sortedBreakpoints = Object.values( breakpoints ).sort( ( a, b ) => {
+		return +a < +b ? -1 : 1;
```
[the remark this answers is not in the document]
Comment (kadamwhite) [reply src/blocks/responsive-container/index.js:42]: It is indeed a syntax hack to treat them as both numeric. 😬

**Review by smccafferty — APPROVED** [review]
Looks good, nice work 👍

**Pull request merged**